go/tools/sizegen/sizegen.go

```diff
@@ -447,12 +549,21 @@ func (sizegen *sizegen) sizeStmtForType(fieldName *jen.Statement, field types.Ty
 		ts := sizegen.getKnownType(node)
 		if ts.pod || !ts.local {
 			if alloc {
-				if !ts.local {
-					log.Printf("WARNING: size of external type %s cannot be fully calculated", node)
+				var stmts []jen.Code
+				if node.String() == "math/big.Int" {
+					// This type is not accessible, but with the given
+					// accessors we can compute a proper size.
```

Comment:
Custom handling here for `big.Int` for which we can more accurately compute the size.